src/lib/utils/loadstor.h

```diff
@@ -83,424 +121,558 @@ inline constexpr uint64_t make_uint64(
+ * The caller provided some integer values but did not provide the output
+ * container. Let's create one for them, fill it with one of the overloads above
+ * and return it. This will default to a std::array if the caller did not
+ * specify the desired output container type.
+ *
+ * @param ins some words that should be stored
+ * @return a container of bytes that contains the stored words
+ */
+template <Endianness endianness, typename OutR, unsigned_integralish... Ts>
+   requires all_same_v<Ts...>
+inline constexpr auto store_any(Ts... ins) {
+   return store_any<endianness, OutR>(std::array{ins...});
+}
+
+//
+// Legacy store functions that work on raw pointers and arrays
+//
+
+/**
+ * Store a single unsigned integer into a raw pointer
+ * @param in the input unsigned integer
+ * @param out the byte array to write to
+ */
+template <Endianness endianness, typename InT, unsigned_integralish T>
+   requires(std::same_as<AutoDetect, InT> || std::same_as<T, InT>)
+inline constexpr void store_any(T in, uint8_t out[]) {
+   // asserts that *out points to enough bytes to write into
+   store_any<endianness, InT>(in, std::span<uint8_t, sizeof(T)>(out, sizeof(T)));
+}
+
+/**
+ * Store many unsigned integers words into a raw pointer
+ * @param ins a arbitrary-length parameter list of unsigned integers to be stored
+ * @param out the byte array to write to
+ */
+template <Endianness endianness, typename InT, unsigned_integralish T0, unsigned_integralish... Ts>
+   requires(std::same_as<AutoDetect, InT> || std::same_as<T0, InT>) && all_same_v<T0, Ts...>
+inline constexpr void store_any(uint8_t out[], T0 in0, Ts... ins) {
+   constexpr auto bytes = sizeof(in0) + (sizeof(ins) + ... + 0);
    // asserts that *out points to the correct amount of memory
-   store_le(std::span<uint8_t, bytes>(out, bytes), ins...);
+   store_any<endianness, T0>(std::span<uint8_t, bytes>(out, bytes), in0, ins...);
 }
 
+}  // namespace detail
+
 /**
-* Store a big-endian unsigned integer
-* @param in the input unsigned integer
-* @return a byte array holding the integer value in big-endian byte order
-*/
-template <std::unsigned_integral T>
-inline constexpr auto store_be(T in) {
-   std::array<uint8_t, sizeof(T)> out;
-   store_be(in, out);
-   return out;
+ * Store "something" in little endian byte order
+ * See the documentation of this file for more details.
+ */
+template <typename ModifierT = detail::AutoDetect, typename... ParamTs>
+inline constexpr auto store_le(ParamTs&&... params) {
+   return detail::store_any<detail::Endianness::Little, ModifierT>(std::forward<ParamTs>(params)...);
 }
 
 /**
-* Store a little-endian unsigned integer
-* @param in the input unsigned integer
-* @return a byte array holding the integer value in little-endian byte order
-*/
-template <std::unsigned_integral T>
-inline constexpr auto store_le(T in) {
-   std::array<uint8_t, sizeof(T)> out;
-   store_le(in, out);
-   return out;
+ * Store "something" in little endian byte order
```

Comment:
```suggestion
 * Store "something" in big endian byte order
```

copy & paste screwup.